ghauth.js

```diff
@@ -35,123 +41,174 @@ function newlineify (len, str) {
   return s
 }
 
-async function createAuth (options) {
-  const reqOptions = {
-    headers: {
-      'X-GitHub-OTP': options.otp || null,
-      'User-Agent': options.userAgent || defaultUA,
-      'Content-type': 'application/json'
-    },
-    method: 'post',
-    auth: `${options.user}:${options.pass}`
-  }
-  const authUrl = options.authUrl || defaultAuthUrl
-  const currentDate = new Date().toJSON()
+function delay (s) {
+  const ms = s * 1000
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
 
+async function hyperquestJson (url, query, reqOptions) {
```

Comment:
If we switch to node-fetch or something more promise, we could clean this up more.